data/trie/trieStorageManager.go

```diff
@@ -0,0 +1,326 @@
+package trie
+
+import (
+	"os"
+	"path"
+	"strconv"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/mock"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+// trieStorageManager manages all the storage operations of the trie (commit, snapshot, checkpoint, pruning)
+type trieStorageManager struct {
+	db            data.DBWriteCacher
+	pruningBuffer [][]byte
+
+	snapshots       []storage.Persister
+	snapshotId      int
+	snapshotDbCfg   config.DBConfig
+	snapshotsBuffer snapshotsBuffer
+
+	dbEvictionWaitingList data.DBRemoveCacher
+	storageOperationMutex sync.RWMutex
+}
+
+// NewTrieStorageManager creates a new instance of trieStorageManager
+func NewTrieStorageManager(db data.DBWriteCacher, snapshotDbCfg config.DBConfig, ewl data.DBRemoveCacher) (*trieStorageManager, error) {
```

Comment:
send the `config.DBConfig` as a pointer and check if it is nil

Reply:
Done.